packages/react-core/src/components/TextInputGroup/__tests__/TextInputGroupMain.test.tsx

```diff
@@ -0,0 +1,278 @@
+    render(
+      <TextInputGroupMain hint="Test" type="search">
+        Test
+      </TextInputGroupMain>
+    );
+
+    const hintInput = screen.getByRole('textbox', { hidden: true });
+
+    expect(hintInput).toBeDisabled();
+  });
+
+  it('renders the hint input as aria-hidden', () => {
+    // we set the type of the main input to search here so that we can accurately target the hint input
+    render(
+      <TextInputGroupMain hint="Test" type="search">
+        Test
+      </TextInputGroupMain>
+    );
+
+    const hintInput = screen.getByRole('textbox', { hidden: true });
+
+    expect(hintInput).toHaveAttribute('aria-hidden', 'true');
+  });
+
+  it('passes the hint prop to the hint input as its value', () => {
+    // we set the type of the main input to search here so that we can accurately target the hint input
+    render(
+      <TextInputGroupMain hint="value text" type="search">
+        Test
+      </TextInputGroupMain>
+    );
+
+    const hintInput = screen.getByRole('textbox', { hidden: true });
+
+    expect(hintInput).toHaveAttribute('value', 'value text');
+  });
+
+  it('does not call onChange callback when the input does not change', () => {
+    const onChangeMock = jest.fn();
+
+    render(<TextInputGroupMain onChange={onChangeMock}>Test</TextInputGroupMain>);
+
+    expect(onChangeMock).not.toHaveBeenCalled();
+  });
+
+  it('calls the onChange callback when the input changes', () => {
+    const onChangeMock = jest.fn();
+
+    render(<TextInputGroupMain onChange={onChangeMock}>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+    userEvent.type(input, 'Test');
+
+    expect(onChangeMock).toHaveBeenCalledTimes(4);
+  });
+
+  it('does not call onFocus callback when the input does not get focus', () => {
+    const onFocusMock = jest.fn();
+
+    render(<TextInputGroupMain onFocus={onFocusMock}>Test</TextInputGroupMain>);
+
+    expect(onFocusMock).not.toHaveBeenCalled();
+  });
+
+  it('calls the onFocus callback when the input is focused', () => {
+    const onFocusMock = jest.fn();
+
+    render(<TextInputGroupMain onFocus={onFocusMock}>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+    userEvent.click(input);
+
+    expect(onFocusMock).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not call onBlur callback when the input does not lose focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<TextInputGroupMain onBlur={onBlurMock}>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+    userEvent.click(input);
+
+    expect(onBlurMock).not.toHaveBeenCalled();
+  });
+
+  it('calls the onBlur callback when the input loses focus', () => {
+    const onBlurMock = jest.fn();
+
+    render(<TextInputGroupMain onBlur={onBlurMock}>Test</TextInputGroupMain>);
+
+    const input = screen.getByRole('textbox');
+    userEvent.click(input);
+    userEvent.click(document.body);
```

Comment:
This is a good idea. I haven't thought to focus out by simply clicking the document body before.